packages/clerk-js/src/ui/common/Billing/PlanAndBillingScreen.tsx

```diff
@@ -0,0 +1,164 @@
+import type { BillingPlanResource } from '@clerk/types';
+import React from 'react';
+
+import { Box, Button, Col, descriptors, Flex, localizationKeys, Text, useLocalizations } from '../../customizables';
+import { Card, Header, ProfileSection, useCardState } from '../../elements';
+import { mqu } from '../../styledSystem';
+import { centsToUnit, getRelativeToNowDateKey, handleError } from '../../utils';
+import { useBillingContext } from './BillingProvider';
+
+type PaymentMethodSectionProps = {
+  currentPlan?: BillingPlanResource | null;
+};
+
+export const PaymentMethodSection = ({ currentPlan }: PaymentMethodSectionProps) => {
+  const { t } = useLocalizations();
+
+  if (!currentPlan) {
+    return null;
+  }
+
+  return (
+    <ProfileSection.Root
+      title={localizationKeys('billing.paymentMethodSection.title')}
+      id='paymentMethod'
+      centered={false}
+      sx={{ [mqu.md]: { alignItems: 'flex-start' } }}
+    >
+      <ProfileSection.Item id='paymentMethod'>
+        {currentPlan.paymentMethod ? (
+          <Flex sx={{ flexDirection: 'column' }}>
+            <Text>•••• {currentPlan.paymentMethod.card.last4}</Text>
+            <Text sx={t => ({ color: t.colors.$colorTextSecondary, fontSize: t.fontSizes.$sm })}>
+              {t(localizationKeys('billing.paymentMethodSection.expires'))} {currentPlan.paymentMethod.card.expMonth}/
+              {currentPlan.paymentMethod.card.expYear}
+            </Text>
+          </Flex>
+        ) : (
+          <ProfileSection.ArrowButton
+            id='paymentMethod'
+            localizationKey={localizationKeys('billing.paymentMethodSection.primaryButton')}
+          />
+        )}
+      </ProfileSection.Item>
+    </ProfileSection.Root>
+  );
+};
+
+const CurrentPlanSection = ({ currentPlan }: { currentPlan?: BillingPlanResource | null }) => {
```

Comment:
Can we reuse `PaymentMethodSectionProps` here ?

Reply:
Good point! Actually I can omit the props from the `CurrentPlanSection` as we can consume the currentPlan from the `BillingContext`